internal/eva/login.go

```diff
@@ -0,0 +1,50 @@
+package eva
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+)
+
+const (
+	path = "/api/core/Login"
+)
+
+type LoginCredentials struct {
+	Username string `json:"username"`
+	Password string `json:"password"`
+}
+
+type LoginResponse struct {
+	AuthenticationToken string `json:"AuthenticationToken"`
+}
+
+func (c *Client) Login(ctx context.Context, credentials LoginCredentials) error {
+
+	resp, err := c.Client.R().
+		SetBody(credentials).
+		Post(path)
+
+	if err != nil {
+		tflog.Error(ctx, "An network error ocurred.", err)
+
+		return err
+	}
+
+	if resp.StatusCode() != 200 {
+		tflog.Trace(ctx, "Login failed.", "Status code", resp.StatusCode(), "body", resp.String())
+
+		return errors.New("Login failed.")
+	}
+
+	var loginResponse LoginResponse
+	if err := json.Unmarshal([]byte(resp.Body()), &loginResponse); err != nil {
+		panic(err)
```

Comment:
We shouldn't `panic` here. Otherwise Terraform won't be able to be able to handle the error. Better to bubble it up.

Reply:
This was me being lazy 😂 will fix it 👍🏼